src/v3/src/transformer/selectAuthenticator/transformSelectAuthenticatorEnroll.test.ts

```diff
@@ -190,4 +213,288 @@ describe('Enroll Authenticator Selector Transformer Tests', () => {
+      .options.buttons[0] as AuthenticatorButtonElement).label).toBe('Email');
+  });
+
+  it('should transform authenticator elemtns when no elements have grace periods', () => {
+    const options = [
+      {
+        label: 'Email',
+        value: 'okta_email',
+      } as unknown as IdxOption,
+      {
+        label: 'Email',
+        value: 'okta_email',
+      } as unknown as IdxOption,
+    ];
+
+    transaction.nextStep = {
+      name: IDX_STEP.SELECT_AUTHENTICATOR_ENROLL,
+      canSkip: isSkippable.mockReturnValue(false)(),
+      inputs: [{
+        name: 'authenticator',
+        options,
+      }],
+    };
+
+    const updatedFormBag = transformSelectAuthenticatorEnroll({
+      transaction, formBag, widgetProps,
+    });
+
+    expect(mockGetAuthenticatorEnrollButtonElementsFn).toBeCalledTimes(2);
+    expect(mockGetAuthenticatorEnrollButtonElementsFn).toBeCalledWith([], 'select-authenticator-enroll', 'ok_pl', undefined);
+    expect(mockGetAuthenticatorEnrollButtonElementsFn).toBeCalledWith(options, 'select-authenticator-enroll', 'ok_pl', undefined);
+    expect(updatedFormBag).toMatchSnapshot();
+    expect(updatedFormBag.uischema.elements.length).toBe(4);
+    expect(updatedFormBag.uischema.elements[0].type).toBe('Title');
+    // @ts-ignore TODO: Add grace period fields to auth-js SDK https://oktainc.atlassian.net/browse/OKTA-848910
+    expect(updatedFormBag.uischema.elements[0].options.content).toBe('oie.select.authenticators.enroll.title');
+    expect(updatedFormBag.uischema.elements[1].type).toBe('Description');
+    // @ts-ignore TODO: Add grace period fields to auth-js SDK https://oktainc.atlassian.net/browse/OKTA-848910
+    expect(updatedFormBag.uischema.elements[1].options.content).toBe('oie.select.authenticators.enroll.subtitle');
+    expect(updatedFormBag.uischema.elements[2].type).toBe('Heading');
+    // @ts-ignore TODO: Add grace period fields to auth-js SDK https://oktainc.atlassian.net/browse/OKTA-848910
+    expect(updatedFormBag.uischema.elements[2].options.content).toBe('oie.setup.required.now');
+    expect(((updatedFormBag.uischema.elements[3] as AuthenticatorButtonListElement)
+      .options.dataSe)).toBe('authenticator-enroll-list');
+    expect(((updatedFormBag.uischema.elements[3] as AuthenticatorButtonListElement)
+      .options.buttons[0] as AuthenticatorButtonElement).options.type).toBe(ButtonType.BUTTON);
+    expect(((updatedFormBag.uischema.elements[3] as AuthenticatorButtonListElement)
+      .options.buttons[0] as AuthenticatorButtonElement).label).toBe('Email');
+  });
+
+  it('should transform authenticator elements when all elements have expired grace periods', () => {
+    const options = [
+      {
+        relatesTo: {
+          gracePeriod: {
+            expiry: '2020-09-27 18:00:00.000.',
+          },
+        },
+      } as unknown as IdxOption,
+      {
+        relatesTo: {
+          gracePeriod: {
+            expiry: '2020-9-27 18:00:00.000.',
```

Comment:
Not sure if the missing `0` was intentional or not

Reply:
Probably not intentional, updating the timestamps to match the rest of the mocks